### PR TITLE
Refactor puma upstart config

### PR DESCRIPTION
**Pull request by nullobject**
This PR refactors the puma upstart deps. The main motivation was to be able to control the number of puma threads/workers from the provision dep.

- Remove the custom `puma upstart config` dep, use the `puma.upstart` template instead.
- Always add `PUMA_THREADS` and `PUMA_WORKERS` environment values.
- Apps can specify `threads` and `workers` as params to the `sinatra app` dep.
- Add bundler dep to counter app.

### CODE REVIEW

File: puma/puma.init.erb [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-
-chdir <%=app_root%>
-setuid <%=user%>
-console log
-env LANG=en_AU.UTF-8
-env RACK_ENV=<%=env%>
-
-script
-  ./bin/puma
-end script
```
Comment (yob) [inline puma/puma.init.erb]: So we don't need a custom upstart config for puma anymore, the standard `upstart` template will work?

Comment (nullobject) [inline puma/puma.init.erb]: Yes, I have tested it on Rackspace. When I initially created the puma upstart template I just cargoculted the unicorn one without giving it much thought. I don't think it's necessary to have another template that doesn't do anything special.

**Comment by yob** [comment]
I haven't tested this, but the diff looks sensible. Once merged, will we try it on staging first?

**Comment by nullobject** [comment]
> I haven't tested this, but the diff looks sensible. Once merged, will we try it on staging first?

Yep, to be verified on staging 😄

**Pull request merged**